python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -375,6 +375,19 @@ def test_generic_hints(self):
         plan = df1.join(df2.hint("broadcast"), "id")._jdf.queryExecution().executedPlan()
         self.assertEqual(1, plan.toString().count("BroadcastHashJoin"))
 
+    # add tests for SPARK-23647 (test more types for hint)
+    def test_extended_hint_types(self):
+        from pyspark.sql import DataFrame
+
+        df = self.spark.range(10e10).toDF("id")
+        such_a_nice_list = ["itworks1", "itworks2", "itworks3"]
```

Comment:
@DylanGuedes, what do we get if `dict` is given? looks not being tested. If that produces a weird result, we can disallow it here for now.

Reply:
@HyukjinKwon I tested locally and you are right, it is kinda weird, so I just removed it.